tracer/src/Datadog.Trace/SignalFx/Metrics/SignalFxMetricExporter.cs

```diff
@@ -116,15 +116,19 @@ private static bool IsTransient(Exception ex)
                 var webExceptionStatus = webException.Status;
                 if (webExceptionStatus != WebExceptionStatus.ProtocolError)
                 {
-                    // these are considered to be transient
                     return TransientExceptionStatuses.Contains(webExceptionStatus);
                 }
 
                 using var response = (HttpWebResponse)webException.Response;
-                return TransientErrorStatusCodes.Contains(response.StatusCode);
+                return IndicatesTransientError((int)response.StatusCode);
             }
 
             return false;
         }
+
+        private static bool IndicatesTransientError(int responseStatusCode)
+        {
+            return TransientClientErrorStatusCodes.Contains(responseStatusCode) || (responseStatusCode >= 500 && responseStatusCode <= 599);
```

Comment:
how about removing the `TransientClientErrorStatusCodes` to reduce the memory allocations? also we would have all status codes in one place (higher cohesion). e.g. something like 

```suggestion
        {
            // following HTTP status codes are considered as a transient error:
            // 408 Request Timeout
            // 429 Too Many Requests
            // 5xx Server Error
            return responseStatusCode == 408 || responseStatusCode == 429 || (responseStatusCode >= 500 && responseStatusCode <= 599);
```